tests/watcher_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	client "k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+const (
+	TEST_GLOBAL_SVC_NAME = "echo-test-svc-global"
+	TEST_NS              = "default"
+	TEST_RETRIES         = 5
+)
+
+func GetEndpintSlices() {
+	var kubeconfig *string
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	}
```

Comment:
This is weird UX. So if a home directory does not exist, we parse the flag? It's usually the other way around. Sounds like we have no way to provide overrides for this. Maybe just keep it simple and always assume a home dir exists.

Reply:
Removing it all together